File: cpp/cpp_1000-/1455.cc
// C++
// leetcode 1455
// https://leetcode.cn/problems/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/
// 从句子里切分出单词，并对每个单词check

class Solution {
public:
    int isPrefixOfWord(string sentence, string word) {
    	int idx = 0, n = sentence.size(), m = word.size();

    	// check one word per loop 
    	for (int i = 0; i < n; i++) {
    		idx++;

    		// check
    		int p = 0;
    		while (i + p < n && p < m) {
    			if (sentence[i + p] == word[p])
    				p++;
    			else break;
    		}
    		if (p == m)
    			return idx;

    		// find position for next word
    		i += p;
    		while(i < n && sentence[i] != ' ') {
    			i++;
    		}
    	}

    	return -1;
    }
};

File: Go/Go_1000-/1455.go
// Go
// leetcode 1455
// https://leetcode.cn/problems/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/

func isPrefixOfWord(sentence string, word string) int {
	n, m, idx := len(sentence), len(word), 0
	for i := 0; i < n; i++ {
		idx++

		p := 0
		for p+i < n && p < m {
			if sentence[i+p] == word[p] {
				p++
			} else {
				break
			}
		}

		if p == m {
			return idx
		}

		i += p
		for i < n && sentence[i] != ' ' {
			i++
		}
	}

	return -1
}